core/app/models/spree/address/name.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module Spree
+  class Address
+    # Provides a value object to help transitioning from legacy
+    # firstname and lastname fields to a unified name field.
+    class Name
+      attr_reader :first_name, :last_name, :full_name
```

Comment:
I think the attribute should be just `name`, since `full_name` is a legacy name. If `Spree::Address::Name.new.name` seems odd (because of the duplication) we could name it `value`.

Reply:
Agree, `value` is a good alternative.

Reply:
👍